analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt

```diff
@@ -80,18 +88,48 @@ private fun getConcludedLicense(pkg: SpdxPackage): SpdxExpression? =
     pkg.licenseConcluded.takeIf { SpdxConstants.isPresent(it) }?.toSpdx()
 
 /**
- * Return a [RemoteArtifact] created from the downloadLocation of the given package [SpdxPackage] if it is a local
- * file, or return an [RemoteArtifact.EMPTY].
- *
- * TODO: Consider also taking "sourceInfo" or "externalRefs" into account.
+ * Return a [RemoteArtifact] for the binary artifact that the [pkg]'s [downloadLocation][SpdxPackage.downloadLocation]
+ * points to. If the download location is a "not present" value, or if it points to a source artifact or a VCS location
+ * instead, return null.
+ */
+internal fun getBinaryArtifact(pkg: SpdxPackage): RemoteArtifact? = getRemoteArtifact(pkg, !pkg.filesAnalyzed)
+
+/**
+ * Return a [RemoteArtifact] for the source artifact that the [pkg]'s [downloadLocation][SpdxPackage.downloadLocation]
+ * points to. If the download location is a "not present" value, or if it points to a binary artifact or a VCS location
+ * instead, return null.
  */
-private fun getSourceArtifact(pkg: SpdxPackage): RemoteArtifact =
-    if (pkg.downloadLocation.startsWith("file:/")) {
-        RemoteArtifact(pkg.downloadLocation, Hash.NONE)
-    } else {
-        RemoteArtifact.EMPTY
+internal fun getSourceArtifact(pkg: SpdxPackage): RemoteArtifact? = getRemoteArtifact(pkg, pkg.filesAnalyzed)
+
+private fun getRemoteArtifact(pkg: SpdxPackage, condition: Boolean): RemoteArtifact? =
+    when {
+        SpdxConstants.isNotPresent(pkg.downloadLocation) -> null
+        !condition -> null
```

Comment:
It is not clear to me, what this `condition` means. Could the parameter be given a more speaking name?

And for my understanding: The function basically checks three conditions. If one of them is fulfilled, it returns *null*. Is this idiomatic Kotlin to express such a construct as a `when` statement? Naively, I would have used an `if` with OR conditions.

Reply:
I've rewritten this in parallel in a different way, please have a look at the new version.

Reply:
This is way better now